bundles/binding/org.openhab.binding.fritzboxtr064/src/main/java/org/openhab/binding/fritzboxtr064/internal/Tr064Comm.java

```diff
@@ -0,0 +1,707 @@
+		_alItemMap = new ArrayList<ItemMap>();
+		init();
+	}
+
+
+	public String get_url() {
+		return _url;
+	}
+
+
+	public void set_url(String _url) {
+		this._url = _url;
+	}
+
+
+	public String get_user() {
+		return _user;
+	}
+
+
+	public void set_user(String _user) {
+		this._user = _user;
+	}
+
+
+	public String get_pw() {
+		return _pw;
+	}
+
+
+	public void set_pw(String _pw) {
+		this._pw = _pw;
+	}
+	
+	/***
+	 * makes sure all values are set properly
+	 * before starting communications
+	 */
+	private void init(){
+		if(_user == null){
+			_user = DEFAULTUSER;  
+		}
+		if(_httpClient == null){
+			_httpClient = createTr064HttpClient(_url); //create http client used for communication
+		}
+		if(_alServices.isEmpty()){ // no services are known yet?
+			readAllServices(); //can be done w/out item mappings and w/out auth
+		}
+		if(_alItemMap.isEmpty()){ // no mappings present yet?
+			generateItemMappings();
+		}
+	}
+	
+	
+	
+
+	/***
+	 * Fetches a specific value from FritzBox
+	 * 	 
+	 * 
+	 * @param request string from config including the command and optional parameters
+	 * @return parsed value
+	 */
+	public String getTr064Value(String request){
+		String value = null;
+		
+		//extract itemCommand from request
+		String[] itemConfig = request.split(":");
+		String itemCommand = itemConfig[0]; //command is always first
+		
+		//search for proper item Mapping
+		ItemMap itemMap = determineItemMappingByItemCommand(itemCommand); 
+		
+		if(itemMap == null){
+			logger.warn("No item mapping found for "+request+". function not implemented by fbox?");
+			return "";
+		}
+		
+		//determine which url etc. to connect to for accessing required value
+		Tr064Service tr064service = determineServiceByItemMapping(itemMap);
+		
+		//construct soap Body which is added to soap msg later
+		SOAPBodyElement bodyData = null; //holds data to be sent to fbox
+		try {
+			MessageFactory mf = MessageFactory.newInstance();
+			SOAPMessage msg = mf.createMessage(); //empty message
+			SOAPBody body = msg.getSOAPBody(); //std. SAOP body
+			QName bodyName = new QName(tr064service.get_serviceType(), itemMap.get_readServiceCommand(), "u"); //header for body element
+			bodyData = body.addBodyElement(bodyName);
+			//only if input parameter is present
+			if(itemConfig.length > 1){
+				String dataInValue = itemConfig[1];
+				QName dataNode = new QName(itemMap.get_readDataInName()); //service specific node name
+				SOAPElement beDataNode = bodyData.addChildElement(dataNode);
+				//if input is mac address, replace "-" with ":" as fbox wants
+				if(itemMap.get_itemCommand().equals("maconline")){
+					dataInValue = dataInValue.replaceAll("-", ":");
+				}
+				beDataNode.addTextNode(dataInValue); //add data which should be requested from fbox for this service
+				//logger.debug(soapToString(msg));
+			}
+			
+			
+		} catch (Exception e) {
+			logger.error("Error constructing request SOAP msg for getting parameter");
```

Comment:
decrease to WARN() since ERROR should only be set if the framework stability is at risk